src/building-blocks/CardFieldSet.tsx

```diff
@@ -28,24 +29,42 @@ class CardFieldSet extends Component<ICardFieldSetProps> {
     return fillInFieldSet(this.props.fieldSet);
   }
 
+  private renderLegend () {
+    const fieldSet = this.fieldSet;
+
+    if (isFieldSetSimple(fieldSet)) { return null; }
+    const { legend, tooltip } = fieldSet;
+
+    if (!legend) { return null; }
+
+    return (
+      <h3>
+        {tooltip
```

Comment:
Just wondering why this is not more like react nested components, like instead of having a function called renderWithToolTip not having something like:
```javascript
return (
   { shouldRenderTooltip ? <ToolTip> <Legend /> </ToolTip> : <Legend /> }
);
```